open-hackathon-server/src/hackathon/hack/hackathon_manager.py

```diff
@@ -126,39 +127,83 @@ def get_hackathon_list(self, args):
+        # query = Hackathon.query
+        # if status:
+        #     query = query.filter(Hackathon.status == status)
+        # if name:
+        #     query = query.filter(Hackathon.name.like("%" + name + "%"))
+
+        # if order_by == "create_time":
+        #     query = query.order_by(Hackathon.create_time.desc())
+        # elif order_by == "event_start_time":
+        #     # all started and coming hackathon-activities would be shown based on event_start_time.
+        #     query = query.order_by(Hackathon.event_start_time.desc())
+
+        #     # just coming hackathon-activities would be shown based on event_start_time.
+        #     # query = query.order_by(Hackathon.event_start_time.asc()).filter(Hackathon.event_start_time > self.util.get_now())
+        # elif order_by == "registered_users_num":
+        #     # hackathons with zero registered users would not be shown.
+        #     query = query.join(HackathonStat).order_by(HackathonStat.count.desc())
+        # else:
+        #     query = query.order_by(Hackathon.id.desc())
+
+        # # perform db query with pagination
+        # pagination = self.db.paginate(query, page, per_page)
+
+        # # check whether it's anonymous user or not
+        # user = None
+        # if self.user_manager.validate_login():
+        #     user = g.user
+
+        # def func(hackathon):
+        #     return self.__get_hackathon_detail(hackathon, user)
+
+        # # return serializable items as well as total count
+        # return self.util.paginate(pagination, func)
+
+        query = Hackathon.objects()
         if status:
-            query = query.filter(Hackathon.status == status)
+            query = query.filter(status=status)
         if name:
-            query = query.filter(Hackathon.name.like("%" + name + "%"))
+            query = query.filter(name__contains=name)
 
         if order_by == "create_time":
-            query = query.order_by(Hackathon.create_time.desc())
+            query = query.order_by('-create_time')
         elif order_by == "event_start_time":
             # all started and coming hackathon-activities would be shown based on event_start_time.
-            query = query.order_by(Hackathon.event_start_time.desc())
+            query = query.order_by('-event_start_time')
 
             # just coming hackathon-activities would be shown based on event_start_time.
             # query = query.order_by(Hackathon.event_start_time.asc()).filter(Hackathon.event_start_time > self.util.get_now())
         elif order_by == "registered_users_num":
             # hackathons with zero registered users would not be shown.
-            query = query.join(HackathonStat).order_by(HackathonStat.count.desc())
+            pass #TODO 
+            # query = query.join(HackathonStat).order_by(HackathonStat.count.desc())
         else:
-            query = query.order_by(Hackathon.id.desc())
+            query = query.order_by('-id')
+
+        # TO CHANGE, no pagination
```

Comment:
Please update the todo line to indicate that we actually need to implement `stat` related features